### PR TITLE
Fix navigation flow when popping to some controller before rootVC

### PR DESCRIPTION
**Pull request by LukasHromadnik**
Currently there is a problem with the following flow:

```
              push               push
RootVC1 [FC1] ---> RootVC2 [FC2] ---> VC [FC2]
```

When we call `popToRootViewController` on `FC2` the flow is not stopped!

In this PR I addressed this issue by checking if the current `rootViewController` is presented in `navigationController.viewControllers` which changes we can observer in `navigationController(didShowViewController)`.

#### Checklist
- [x] Added tests (if applicable)

### CODE REVIEW

**Review by olejnjak — APPROVED** [review]

File: ACKategories-iOSTests/FlowCoordinator/Dummies.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    func push(_ vc: UIViewController) {
+        navigationController?.pushViewController(vc, animated: false)
+    }
+    
+    func present(_ vc: UIViewController) {
+        rootViewController.present(vc, animated: false)
+    }
+}
+
+class NavigationFC: Base.FlowCoordinatorNoDeepLink {
```
Comment (fortmarek) [inline ACKategories-iOSTests/FlowCoordinator/Dummies.swift:21]: nitpick: I'd mark these final

File: ACKategories-iOSTests/FlowCoordinator/Dummies.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        let rootVC = UIViewController()
+        let navVC = UINavigationController(rootViewController: rootVC)
+        rootViewController = rootVC
+        navigationController = navVC
+        
+        window.rootViewController = navigationController
+        window.makeKeyAndVisible()
+        _ = navVC.view
+        
+        super.start(in: window)
```
Comment (fortmarek) [inline ACKategories-iOSTests/FlowCoordinator/Dummies.swift:32]: shouldn't we call `super` as the first thing in this method? (same applies to overrides below)

File: ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    // MARK: - Tests
+
+    func testNavigationSubflowIsStopped() {
+        let fc = NavigationFC()
+        fc.start(in: window)
+        _ = fc.rootViewController.view
+        
+        let childFC = InnerNavigationFC()
+        fc.addChild(childFC)
+        childFC.start(with: fc.navigationController!)
```
Comment (fortmarek) [inline ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift:37]: I'd use `XCTUnwrap(...)`

File: ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        _ = fc.rootViewController.view
+
+        let exp = expectation(description: "did present")
+        let childFC = PresentFC { exp.fulfill() }
+        fc.addChild(childFC)
+        childFC.start(from: fc.rootViewController)
+        _ = childFC.rootViewController.view
+        wait(for: [exp], timeout: 1)
+
+        XCTAssertNotNil(fc.rootViewController!.presentedViewController)
```
Comment (fortmarek) [inline ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift:115]: the force unwrap here is unnecessary

File: ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+        let exp2 = expectation(description: "did present")
+        let child2FC = PresentFC { exp2.fulfill() }
+        childFC.addChild(child2FC)
+        child2FC.start(from: childFC.rootViewController)
+        _ = child2FC.rootViewController.view
+        wait(for: [exp2], timeout: 1)
+
+        XCTAssertNotNil(childFC.rootViewController!.presentedViewController)
```
Comment (fortmarek) [inline ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift:124]: the force unwrap here is unnecessary, too

**Review by fortmarek — APPROVED** [review]
Just a few nitpicks - love to see those tests ❤️

File: ACKategories-iOSTests/FlowCoordinator/Dummies.swift (hunk 1 above)
Comment (LukasHromadnik) [reply ACKategories-iOSTests/FlowCoordinator/Dummies.swift]: 👍

File: ACKategories-iOSTests/FlowCoordinator/Dummies.swift (hunk 2 above)
Comment (LukasHromadnik) [reply ACKategories-iOSTests/FlowCoordinator/Dummies.swift]: 👍

File: ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift (hunk 3 above)
Comment (LukasHromadnik) [reply ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift]: 👍

File: ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift (hunk 4 above)
Comment (LukasHromadnik) [reply ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift]: 🙈 probably some copy-paste

File: ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift (hunk 5 above)
Comment (LukasHromadnik) [reply ACKategories-iOSTests/FlowCoordinator/FlowCoordinatorTests.swift]: 👍

**Comment by LukasHromadnik** [comment]
All comments should be resolved now

**Review by janmisar — APPROVED** [review]

**Pull request merged**